sympy/core/function.py

```diff
@@ -428,7 +428,7 @@ def __new__(cls, *args, **options):
         if not evaluate or not isinstance(result, cls):
```

Comment:
That would certainly be more compact and efficient.

One other aspect to consider is what should happen if evaluation is requested and no arguments are present. According to the comments on `_should_evalf` the original intention was to force `evalf` with sufficient precision **if and only if** all arguments are `real`. If there are no arguments all arguments are real by default, does that mean we have to enforce evalf in such a case? is it even possible? What would be the precision? Or does it perhaps happen automatically? I'm not familiar enough with the numeric evaluation process to hazard a guess.

On the other hand we are talking about quite the edge case here and I some how doubt that it is used all that often.